docs/content/docs/dev/configuration/maven.md

```diff
@@ -0,0 +1,175 @@
+For projects that are not set up from those templates, we recommend adding the Maven Shade Plugin to
+build the application jar with all required dependencies.
+
+**Important:** Note that all these core API dependencies should have their scope set to [*provided*](https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#dependency-scope). This means that
+they are needed to compile against, but that they should not be packaged into the project's resulting
+application JAR file. If not set to *provided*, the best case scenario is that the resulting JAR
+becomes excessively large, because it also contains all Flink core dependencies. The worst case scenario
+is that the Flink core dependencies that are added to the application's JAR file clash with some of
+your own dependency versions (which is normally avoided through inverted classloading).
+
+To correctly package the dependencies into the application JAR, the Flink API dependencies must 
+be set to the *compile* scope.
+
+## Packaging the application
+
+Depending on your use case, you may need to package your Flink application in different ways before it
+gets deployed to a Flink environment. 
+
+If you want to create a JAR for a Flink Job and use only Flink dependencies without any third-party 
+dependencies (i.e. using the filesystem connector with JSON format), you do not need to create an 
+uber/fat JAR or shade any dependencies.
+
+If you want to create a JAR for a Flink Job and use external dependencies not built into the Flink 
+distribution, you can either add them to the classpath of the distribution or shade them into your 
+uber/fat application JAR.
+
+With the generated uber/fat JAR, you can submit it to a local or remote cluster with:
+
+```sh
+bin/flink run -c org.example.MyJob myFatJar.jar
+```
+
+To learn more about how to deploy Flink jobs, check out the [deployment guide]({{< ref "docs/deployment/cli" >}}).
+
+## Template for creating an uber/fat JAR with dependencies
+
+To build an application JAR that contains all dependencies required for declared connectors and libraries,
+you can use the following shade plugin definition:
+
+```xml
+<build>
+    <plugins>
+        <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-shade-plugin</artifactId>
+            <version>3.2.4</version>
+            <executions>
+                <execution>
+                    <phase>package</phase>
+                    <goals>
+                        <goal>shade</goal>
+                    </goals>
+                    <configuration>
+                        <artifactSet>
+                            <excludes>
+                                <exclude>com.google.code.findbugs:jsr305</exclude>
+                                <exclude>org.slf4j:*</exclude>
+                                <exclude>log4j:*</exclude>
```

Comment:
This should be unnecessary because all Flink dependencies have a provided dependency on slf4j and log4j.
Also, the log4j groupId is outdated (`org.apache.logging.log4j` would be the correct choice).